### PR TITLE
Torch2

### PR DESCRIPTION
**Pull request by hxyue1**
## :microscope: Background

- This is to upgrade neuralprophet to use pytorch 2.0 and address #1229

## :crystal_ball: Key changes

- Update the pyproject.toml file to use pytorch 2.0 and add optional CPU dependencies.

## :clipboard: Review Checklist
- [X] I have performed a self-review of my own code.
- [] I have commented my code, added docstrings and data types to function definitions.
- [] I have added pytests to check whether my feature / fix works.

Please make sure to follow our best practices in the [Contributing guidelines](https://github.com/ourownstory/neural_prophet/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -31,8 +31,12 @@ jobs:
           python-version: ${{ matrix.python-version }}
       - name: Install Poetry
         uses: snok/install-poetry@v1
+      - name: Generate lock file
+        run: poetry lock
+      - name: Update lock file
+        run: poetry update
```
Comment (noxan) [inline .github/workflows/ci.yml:37]:
```suggestion
```
The lockfile should not be automatically updated by our CI job, but rather locally - else we lose the benefits of locked dependencies from poetry.

Best update the lockfile locally - without updating any of the other dependencies with `poetry lock --no-update`

File: .github/workflows/ci.yml [hunk 2]
```diff
@@ -56,6 +60,10 @@ jobs:
         uses: abatilo/actions-poetry@v2
       - name: Setup Pandoc
         uses: r-lib/actions/setup-pandoc@v2
+      - name: Generate lock file
+        run: poetry lock
+      - name: Update lock file
+        run: poetry update
```
Comment (noxan) [inline .github/workflows/ci.yml:66]:
```suggestion
```

see above

File: .github/workflows/linters.yml [hunk 3]
```diff
@@ -84,6 +84,10 @@ jobs:
       - name: Setup poetry
         if: steps.poetry-cache.outputs.cache-hit != 'true'
         uses: snok/install-poetry@v1
+      - name: Generate lock file
+        run: poetry lock
+      - name: Update lock file
+        run: poetry update
```
Comment (noxan) [inline .github/workflows/linters.yml:90]:
```suggestion
```
see above

File: pyproject.toml [hunk 4]
```diff
@@ -29,16 +29,28 @@ kaleido = "0.2.1"
 plotly-resampler = "^0.8.3.1"
 pytorch-lightning = "^1.9.4"
 tensorboard = "^2.11.2"
-torch = "^1.13.1"
+torch = "^2.0.0"
```
Comment (noxan) [inline pyproject.toml:32]:
```suggestion
torch = "^2.0.1"
```
there is torch 2.0.1 available already

**Review by noxan — CHANGES_REQUESTED** [review]

**Comment by hxyue1** [comment]
Unfortunately this fails on windows with python 3.8 for some reason, but is fine in all other scenarios. The flake8 and pyright jobs are also failing but I haven't made changes to any python code so I'm not sure why that's the case.

**Comment by noxan** [comment]
@hxyue1, thanks for the update and your work. To better understand the different problems you encountered, I drafted #1404.

I am not a big fan of forcing the `cpu` version of Pytorch - see my comment https://github.com/ourownstory/neural_prophet/issues/1229#issuecomment-1681393452 - let's come up with a plan there and then get one of the pull requests merged.

**Comment by leoniewgnr** [comment]
@hxyue1 I will close this for now because #1404 fixes it, but feel free to open it again if you need it.

**Pull request closed without merging**